src/transformers/generation/utils.py

```diff
@@ -1288,6 +1288,10 @@ def generate(
                 model_kwargs=model_kwargs,
                 device=inputs_tensor.device,
             )
+
+            # conditional generation for multi-modal models.
+            if "input_ids" in model_kwargs and model_input_name == "pixel_values":
+                input_ids = torch.cat([input_ids, model_kwargs.pop("input_ids")], dim=-1)
```

Comment:
Uhmmm this seems to be the same logic as below (`input_ids = inputs_tensor if model_input_name == "input_ids" else model_kwargs.pop("input_ids")`), which is applied on encoder-decoder models.

Perhaps instead of adding these lines, we can remove the `else:` below?